### PR TITLE
Added Planning your environment according to object limits content for 4.0

### PR DESCRIPTION
**Pull request by ahardin-rh**
This includes content for:

- https://jira.coreos.com/browse/OSDOCS-42
- https://jira.coreos.com/browse/OSDOCS-50

Preview build:
http://file.rdu.redhat.com/~ahardin/11142018/planning-environment-according-to-object-limits/scalability_and_performance/planning-your-environment-according-to-object-limits.html

Dev review tracker: https://jira.coreos.com/browse/ATS-183

### CODE REVIEW

**Comment by ahardin-rh** [comment]
@chaitanyaenr  PTAL and let me know if this all looks accurate for 4.0. Thanks!

**Review by chaitanyaenr — APPROVED** [review]
/lgtm

**Comment by ahardin-rh** [comment]
@openshift/team-documentation Please peer review :heart_decoration:  No new content for 4.0. The new limits for 4.0  will be updated when those metrics are available; right now I have them marked as TBD.

File: modules/how-to-plan-your-environment-according-to-application-requirements.adoc [hunk 1]
```diff
@@ -0,0 +1,71 @@
+// Module included in the following assemblies:
+//
+// * scalability_and_performance/planning-your-environment-according-to-object-limits.adoc
+
+[id='how-to-plan-according-to-application-requirements']
```
Comment (kalexand-rh) [inline modules/how-to-plan-your-environment-according-to-application-requirements.adoc]: All the anchors in modules need `{context}` in them, like [id='how-to-plan-according-to-application-requirements-{context}']

File: scalability_and_performance/planning-your-environment-according-to-object-limits.adoc [hunk 2]
```diff
@@ -0,0 +1,30 @@
+[id='planning-your-environment-according-to-object-limits']
+= Planning your environment according to object limits
+{product-author}
```
Comment (kalexand-rh) [inline scalability_and_performance/planning-your-environment-according-to-object-limits.adoc]: Please replace lines 3-10 with "include::modules/common-attributes.adoc[]"

File: scalability_and_performance/planning-your-environment-according-to-object-limits.adoc [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+:experimental:
+:toc: macro
+:toc-title:
+:prewrap!:
+:context: object limits
+
+toc::[]
+
+
+Consider the following object limits when you plan your {product-title} cluster.
```
Comment (kalexand-rh) [inline scalability_and_performance/planning-your-environment-according-to-object-limits.adoc]: Before GA, I think you'll need another paragraph or two about the difference in planning between BYO and next-gen clusters. Some of the modules might only apply to BYO.

File: modules/how-to-plan-your-environment-according-to-cluster-limits.adoc [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+node:
+
+----
+Maximum Pods per Cluster / Expected Pods per Node = Total Number of Nodes
+----
+
+The number of pods expected to fit on a node is dependent on the application
+itself. Consider the application's memory, CPU, and storage requirements.
+
+.Example Scenario
```
Comment (kalexand-rh) [inline modules/how-to-plan-your-environment-according-to-cluster-limits.adoc]: s/Scenario/scenario

File: modules/openshift-cluster-limits.adoc [hunk 5]
```diff
@@ -0,0 +1,82 @@
+// Module included in the following assemblies:
+//
+// * scalability_and_performance/planning-your-environment-according-to-object-limits.adoc
+
+[id='cluster-limits']
+= {product-title} cluster limits
+
+[options="header",cols="5*"]
+|===
+| Limit Type |3.9 Limit |3.10 Limit |3.11 Limit |4.0 Limit
```
Comment (kalexand-rh) [inline modules/openshift-cluster-limits.adoc]: I think the table headings should be sentence case.

File: modules/how-to-plan-your-environment-according-to-application-requirements.adoc [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+= How to plan your environment according to application requirements
+
+Consider an example application environment:
+
+[options="header",cols="5"]
+|===
+|Pod Type |Pod Quantity |Max Memory |CPU Cores |Persistent Storage
```
Comment (kalexand-rh) [inline modules/how-to-plan-your-environment-according-to-application-requirements.adoc]: I think the table headings should be sentence case.

File: modules/how-to-plan-your-environment-according-to-application-requirements.adoc [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+|apache
+|100
+|500MB
```
Comment (kalexand-rh) [inline modules/how-to-plan-your-environment-according-to-application-requirements.adoc]: Unless these are specific variables, you should have a space between the number and unit, eg "500 MB."

File: modules/openshift-cluster-limits.adoc [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+|TBD
+
+| Number of namespaces
+| 10,000
+| 10,000
+| 10,000
+|TBD
+
+| Number of builds: Pipeline Strategy
+| 10,000 (Default pod RAM 512Mi)
```
Comment (kalexand-rh) [inline modules/openshift-cluster-limits.adoc]: I'd make these all "512 Mi"

**Review by kalexand-rh — COMMENTED** [review]
I have a couple of suggestions and a request for later. Otherwise, LGTM!

**Pull request merged**